### PR TITLE
fix: topology edge filter support

### PR DESCRIPTION
**Pull request by itssharmasandeep**

### CODE REVIEW

File: projects/observability/src/shared/dashboard/data/graphql/topology/topology-data-source.model.ts [hunk 1]
```diff
@@ -22,6 +23,7 @@ import {
 } from '../../../../graphql/request/handlers/entities/query/topology/entity-topology-graphql-query-handler.service';
 import { GraphQlDataSourceModel } from '../graphql-data-source.model';
 import { TopologyMetricsData, TopologyMetricsModel } from './metrics/topology-metrics.model';
+import { AttributeExpression, GraphQlFieldFilter } from '../../../../../public-api';
```
Comment (anandtiwary) [inline projects/observability/src/shared/dashboard/data/graphql/topology/topology-data-source.model.ts:26]: please fix the imports

File: projects/observability/src/shared/dashboard/data/graphql/topology/topology-data-source.model.ts [hunk 2]
```diff
@@ -90,20 +100,25 @@ export class TopologyDataSourceModel extends GraphQlDataSourceModel<TopologyData
       metricSpecifications: this.getAllMetricSpecifications(this.edgeMetricsModel)
     };
 
-    return this.query<EntityTopologyGraphQlQueryHandlerService>(
-      filters => ({
+    return this.query<EntityTopologyGraphQlQueryHandlerService>(filters => {
+      const topologyFilters = this.getTopologyFilters(filters as GraphQlFieldFilter[]);
```
Comment (anandtiwary) [inline projects/observability/src/shared/dashboard/data/graphql/topology/topology-data-source.model.ts:104]: we should do a check before this cast.

File: projects/observability/src/shared/dashboard/data/graphql/topology/topology-data-source.model.ts [hunk 3]
```diff
@@ -160,3 +214,13 @@ export interface TopologyData {
...
   edgeMetrics: TopologyMetricsData;
 }
+
+export interface TopologyEdgeFilterConfig {
+  entityType: string;
+  fields: string[];
+}
+
+interface TopologyFilters {
+  forNode: GraphQlFieldFilter[];
```
Comment (anandtiwary) [inline projects/observability/src/shared/dashboard/data/graphql/topology/topology-data-source.model.ts:224]: could we just call it `nodes` and `edges`? The <for> in the variable name gives it an impression that we are writing a for loop

Comment (itssharmasandeep) [reply projects/observability/src/shared/dashboard/data/graphql/topology/topology-data-source.model.ts]: Done

File: projects/observability/src/shared/dashboard/data/graphql/topology/topology-data-source.model.ts (hunk 2 above)
Comment (itssharmasandeep) [reply projects/observability/src/shared/dashboard/data/graphql/topology/topology-data-source.model.ts:105]: What type of check? Checking any fields in the data? The  `GraphQlFIlter` interface only has one function.  and we do not know how to extract data from there.

Comment (anandtiwary) [reply projects/observability/src/shared/dashboard/data/graphql/topology/topology-data-source.model.ts:105]: As in, before casting filters to `GraphQlFieldFilter`, we should check if it has properties of GraphQlFieldFilter or not. There can be Id Filters in the array that may break this method at run time.

Comment (itssharmasandeep) [reply projects/observability/src/shared/dashboard/data/graphql/topology/topology-data-source.model.ts]: Okay, SO I updated the logic, Now we're checking the field filters and if it is not then we're assigning that to node filters for now.

**Review by anandtiwary — APPROVED** [review]

**Review by palbizu — APPROVED** [review]

**Pull request merged**